src/SfxWeb/src/app/views/applications/base/base.component.ts

```diff
@@ -1,17 +1,18 @@
-import { Component, ElementRef, OnInit } from '@angular/core';
+import { Component, ElementRef, Injector } from '@angular/core';
 import { ITab } from 'src/app/shared/component/navbar/navbar.component';
 import { TreeService } from 'src/app/services/tree.service';
 import { IdGenerator } from 'src/app/Utils/IdGenerator';
 import { DataService } from 'src/app/services/data.service';
 import { Constants } from 'src/app/Common/Constants';
 import { IBaseView } from '../../BaseView';
+import { ApplicationsBaseControllerDirective } from '../applicationsBase';
 
 @Component({
   selector: 'app-base',
   templateUrl: './base.component.html',
   styleUrls: ['./base.component.scss']
 })
-export class BaseComponent implements OnInit, IBaseView{
+export class BaseComponent extends ApplicationsBaseControllerDirective implements IBaseView {
```

Comment:
I dont think we need IBaseView here? I think that is implemented with BaseController?

Reply:
it's not in BaseController, but since all instances where IBaseView is now referenced is extending BaseControllerDirective I'll move it there.